cartographer/field_types/schema_relationship.py

```diff
@@ -56,7 +56,7 @@ def resource(self, parent_resource, relationship_key):
         if self.id_attribute is not None:
             relationship_model_class = self.resource_registry_entry().get(ResourceRegistryKeys.MODEL)
             model_id = getattr(parent_resource.model, self.id_attribute)
-            if model_id is not None:
+            if model_id is not None and hasattr(relationship_model_class, 'get'):
```

Comment:
mmm great point! current notes i have in the asana task that this is addressing are:

>>> * likely via abstract base class / interface
* also consider non-SQLAlchemy libraries that dont give `SchemaRelationship` and `SchemaAttribute` objects (w/attrs), but rather dictionaries
* likely allow lambdas / functions / etc?
* or do `.get(key)` vs `getattr(key)` checking (via `isinstance(dict)`)
* should multigetcache just straight up be a dependency?